docker-compose.yml

```diff
@@ -0,0 +1,37 @@
+services:
+  backend:
+    container_name: server
+    build:
+      context: .
+      dockerfile: Dockerfile
+    image: studyhub-server
+    depends_on:
+      db:
+        condition: service_healthy
+    ports:
+      - 8080:5000
+      - 5678:5678
+    volumes:
+      - .:/app
+
+  db:
```

Comment:
One thing to call out is that if you want to run SQL commands, do it in this `db` container

Reply:
Okay, i will do the code review tomorrow to see how you set up the Docker containers.

Reply:
Actually, everything LGTM. Can we merge this in, can wait to test out our docker containers!

Reply:
fatal: 'jack/task/dockerize-development-environment' is not a commit and a branch 'task/dockerize-development-environment' cannot be created from it

I don't know why I get this error when I checkout to your branch after I fork your local repo.

Reply:
oh oops, forgot one important command, check the updated comment: https://github.com/Kenttrann2302/StudyHub-Backend/pull/22#discussion_r1183095626

Reply:
@j-ngo828 oh no, i think we ran in the same error again with the depencies: 

2023-05-02 18:38:20 sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) connection to server at "127.0.0.1", port 5432 failed: Connection refused
2023-05-02 18:38:20     Is the server running on that host and accepting TCP/IP connections?

This error message from Docker Desktop

Reply:
Hang on, I'm so confused, i thought the host and post are already correct?

Reply:
Actually, tell me how did you ran into that issue? Give me a reproduction step

Did the error log occurred after you ran `docker compose up -d --build`?

I just built a fresh image and ran the container on my machine and all dependencies works fine.

Reply:
> Hang on, I'm so confused, i thought the host and post are already correct?

Yup all host and ports are correct

Reply:
Yes i did do this: docker compose up -d --build

And I went on docker container to start both the db and the server, the db runs fine but the server exited after 2 seconds and output the log like I showed above.

Reply:
It might be because the `db` image pulled is not specific enough. Give me a sec, I will need to update the `docker-compose.yml` file to pull a more specific `postgres` image

Can you check your `.env` to make sure the port is updated. How about changing the `DB_HOST` to `0.0.0.0` and `DB_PORT` to `5432` in your `.env` file

Reply:
I still got the same error when changing the host: sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) connection to server at "0.0.0.0", port 5432 failed: Connection refused
2023-05-02 19:06:25     Is the server running on that host and accepting TCP/IP connections?

Reply:
Can you add the following line of code just before [line 49 in `app.py`](https://github.com/Kenttrann2302/StudyHub-Backend/pull/22/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R49)
```python
raise ValueError(f"{database_type}://{database_username}:{database_password}@{database_host}:{database_port}/{database_name}")
```

Then rerun the docker compose command again and see the value of the ValueError to check if it's correct, reply to this comment with the value of the connection url passed into the ValueError as well. I'm thinking it has to do with incorrect connection url. Check to make sure your `.env` has `DB_PORT` instead of `PORT` too

Reply:
Is it okay, since it includes my username and password for my psql server?

Reply:
Hm then let's use the username and password in the `.env` file I have above then.

Actually just copy all the variables I have in the `.env` in the topmost comment in the PR (be sure to save a copy of your `.env` somewhere on your laptop before making this change) to your `.env`, make sure to not have any duplicate variables then retries.

Reply:
Okay, I'm feeling very strange that your .env file works, but not for mine???

Reply:
Thank god it works. Wanna hop on a Discord call now and try to reproduce the issue you ran into?

Reply:
sure, no problem